processor/src/main/resources/org/mapstruct/ap/internal/model/ValueMappingMethod.ftl

```diff
@@ -22,8 +22,8 @@
 
     switch ( ${sourceParameter.name} ) {
     <#list valueMappings as valueMapping>
-        case <@writeSource source=valueMapping.source/>: ${resultName} = <@writeTarget target=valueMapping.target/>;
-        break;
+        case <@writeSource source=valueMapping.source/>: <#if unexpectedValueMappingException?? && valueMapping.target?? && valueMapping.target="<THROW_EXCEPTION>">throw new <@includeModel object=unexpectedValueMappingException />( "Unexpected enum constant: " + ${sourceParameter.name} );<#else>${resultName} = <@writeTarget target=valueMapping.target/>;
```

Comment:
I think that this is not entirely correct. If someone uses `ANY_UNMAPPED` or `ANY_REMAINING` then `unexpectedValueMappingException` will be `null` which means that when someone explicitly wants to throw an exception for a mapping, but have a default for the rest it wouldn't work. 

What I would suggest is to enhance the `MappingEntry` to know whether an exception needs to be thrown or not and check that.

In addition to that I would change `defaultTarget` and `nullTarget` to use `MappingEntry` as well and then you will repeat the same logic as for the `valueMappings`.